hibernate-core/src/main/java/org/hibernate/dialect/PostgreSQLDialect.java

```diff
@@ -823,91 +833,7 @@ public void contributeTypes(TypeContributions typeContributions, ServiceRegistry
 
 		if ( getVersion() >= 820 ) {
 			// HHH-9562
-			typeContributions.contributeJdbcTypeDescriptor( PostgresUUIDType.INSTANCE );
-		}
-	}
-
-	private static class PostgresUUIDType implements JdbcTypeDescriptor {
-		/**
-		 * Singleton access
-		 */
-		private static final PostgresUUIDType INSTANCE = new PostgresUUIDType();
-
-		/**
-		 * Postgres reports its UUID type as {@link java.sql.Types#OTHER}.  Unfortunately
-		 * it reports a lot of its types as {@link java.sql.Types#OTHER}, making that
-		 * value useless for distinguishing one SqlTypeDescriptor from another.
-		 * So here we define a "magic value" that is a (hopefully no collisions)
-		 * unique key within the {@link JdbcTypeDescriptorRegistry}
-		 */
-		private static final int JDBC_TYPE_CODE = 3975;
-
-		@Override
-		public int getJdbcType() {
-			return JDBC_TYPE_CODE;
-		}
-
-		@Override
-		public int getJdbcTypeCode() {
-			return getJdbcType();
-		}
-
-		@Override
-		public boolean canBeRemapped() {
-			return true;
-		}
-
-		@Override
-		public <J> BasicJavaDescriptor<J> getJdbcRecommendedJavaTypeMapping(TypeConfiguration typeConfiguration) {
-			return (BasicJavaDescriptor<J>) typeConfiguration.getJavaTypeDescriptorRegistry().resolveDescriptor( UUID.class );
-		}
-
-		@Override
-		public <T> JdbcLiteralFormatter<T> getJdbcLiteralFormatter(JavaTypeDescriptor<T> javaTypeDescriptor) {
-			return null;
-		}
-
-		@Override
-		public <X> ValueBinder<X> getBinder(JavaTypeDescriptor<X> javaTypeDescriptor) {
-			return new BasicBinder<X>( javaTypeDescriptor, this ) {
-				@Override
-				protected void doBind(
-						PreparedStatement st,
-						X value,
-						int index,
-						WrapperOptions wrapperOptions) throws SQLException {
-					st.setObject( index, javaTypeDescriptor.unwrap( value, UUID.class, wrapperOptions ), Types.OTHER );
-				}
-
-				@Override
-				protected void doBind(
-						CallableStatement st,
-						X value,
-						String name,
-						WrapperOptions wrapperOptions) throws SQLException {
-					st.setObject( name, javaTypeDescriptor.unwrap( value, UUID.class, wrapperOptions ), Types.OTHER );
-				}
-			};
-		}
-
-		@Override
-		public <X> ValueExtractor<X> getExtractor(JavaTypeDescriptor<X> javaTypeDescriptor) {
-			return new BasicExtractor<X>( javaTypeDescriptor, this ) {
-				@Override
-				protected X doExtract(ResultSet rs, int position, WrapperOptions wrapperOptions) throws SQLException {
-					return javaTypeDescriptor.wrap( rs.getObject( position ), wrapperOptions );
-				}
-
-				@Override
-				protected X doExtract(CallableStatement statement, int position, WrapperOptions wrapperOptions) throws SQLException {
-					return javaTypeDescriptor.wrap( statement.getObject( position ), wrapperOptions );
-				}
-
-				@Override
-				protected X doExtract(CallableStatement statement, String name, WrapperOptions wrapperOptions) throws SQLException {
-					return javaTypeDescriptor.wrap( statement.getObject( name ), wrapperOptions );
-				}
-			};
```

Comment:
I removed the above content to the private static child class in PostgresUUIDType (i.e. PostgresUUIDJdbcTypeDescriptor). See the code changes in PostgresUUIDType below for details.

After we register PostgresUUIDType as below, we also need to register the magical jdbc sql type into Dialect as above.